pkg/controller/account/ec2.go

```diff
@@ -180,12 +183,77 @@ func (r *ReconcileAccount) InitializeRegion(
 
 	successMsg := fmt.Sprintf("EC2 instance created and terminated successfully in region: %s", region)
 
+	// if vpcID exists then make sure we delete it
+	if sampleVPCID != "" {
+		err = DeleteVpc(reqLogger, awsClient, sampleVPCID)
+		if err != nil {
+			deleteVpcErr := fmt.Sprintf("Unable to delete VPC with ID: %s", sampleVPCID)
+			controllerutils.LogAwsError(reqLogger, deleteVpcErr, nil, err)
+		}
+	}
+
 	// Notify Notifications channel that an instance has successfully been created and terminated and to move on
 	ec2Notifications <- successMsg
 
 	return nil
 }
 
+// GetSubnetID builds the vpc and create subnet
+func GetSubnetID(reqLogger logr.Logger, client awsclient.Client) (string, error) {
```

Comment:
Maybe `GetSubnetID` could still be here and just _look_ for the VPC and Subnet and return the subnet ID if the VPC/Subnet for AAO exist (lookup could use tags) and then this way it would be easier to abstract the creation of the VPC/Subnet out of the ec2 runinstances function, as we could use this both to check if the subnet's already created before creating a new one AND just to get the subnet ID without having to pass it into the function and add another parameter.